integration-tests/src/test/java/com/datastax/oss/driver/mapper/RunTimeAttributesIT.java

```diff
@@ -0,0 +1,255 @@
+    simpleDao = inventoryMapper.simpleDao(sessionRule.keyspace());
+    RuntimeAttributes attributes = buildRunTimeAttributes();
+    simulacronRule.cluster().clearLogs();
+    simpleDao.delete(simple, attributes);
+    ClusterQueryLogReport report = simulacronRule.cluster().getLogs();
+    validateQueryOptions(report.getQueryLogs().get(0));
+  }
+
+  @Test
+  public void should_honor_runtime_attributes_select() {
+    Map<String, Object> params = ImmutableMap.of("pk", simple.getPk());
+    Map<String, String> paramTypes = ImmutableMap.of("pk", "uuid");
+    simulacronRule
+        .cluster()
+        .prime(
+            when(query(
+                    "SELECT pk,data FROM simple WHERE pk=:pk",
+                    Lists.newArrayList(
+                        com.datastax.oss.simulacron.common.codec.ConsistencyLevel.ONE,
+                        com.datastax.oss.simulacron.common.codec.ConsistencyLevel.ANY),
+                    params,
+                    paramTypes))
+                .then(noRows())
+                .delay(1, TimeUnit.MILLISECONDS));
+    CqlSession session = SessionUtils.newSession(simulacronRule);
+    InventoryMapper inventoryMapper =
+        new RunTimeAttributesIT_InventoryMapperBuilder(session).build();
+    simpleDao = inventoryMapper.simpleDao(sessionRule.keyspace());
+
+    RuntimeAttributes attributes = buildRunTimeAttributes();
+    simulacronRule.cluster().clearLogs();
+    simpleDao.findByPk(simple.getPk(), attributes);
+    ClusterQueryLogReport report = simulacronRule.cluster().getLogs();
+    validateQueryOptions(report.getQueryLogs().get(0));
+  }
+
+  private RuntimeAttributes buildRunTimeAttributes() {
+    DefaultRunTimeAttributesBuilder builder = DefaultRunTimeAttributes.builder();
+
+    return builder
+        .withConsistencyLevel(DefaultConsistencyLevel.ANY)
+        .withPageSize(PAGE_SIZE)
+        .withSerialConsistencyLevel(DefaultConsistencyLevel.QUORUM)
+        .withPagingState(ByteBuffer.wrap(PAGING_STATE.getBytes(UTF_8)))
+        .build();
+  }
+
+  private void validateQueryOptions(QueryLog log) {
+
+    Message message = log.getFrame().message;
+    assertThat(message).isInstanceOf(Execute.class);
+    Execute queryExecute = (Execute) message;
+    assertThat(queryExecute.options.consistency)
+        .isEqualTo(DefaultConsistencyLevel.ANY.getProtocolCode());
+    assertThat(queryExecute.options.serialConsistency)
+        .isEqualTo(DefaultConsistencyLevel.QUORUM.getProtocolCode());
+    assertThat(queryExecute.options.pageSize).isEqualTo(PAGE_SIZE);
+    String pagingState = UTF_8.decode(queryExecute.options.pagingState).toString();
+    assertThat(pagingState).isEqualTo(PAGING_STATE);
+  }
+
+  @Mapper
+  public interface InventoryMapper {
+    @DaoFactory
+    RunTimeAttributesIT.SimpleDao simpleDao(@DaoKeyspace CqlIdentifier keyspace);
+  }
+
+  @Dao
+  public interface SimpleDao {
+    @Insert
+    void save(Simple simple, RuntimeAttributes attributes);
```

Comment:
Should we use annotation based mechanism for that as pointed in a ticket?
What if someone will want to give that save method Consistency Level `ANY`?
In the current mechanism, he will need to remember about passing that RuntimeAttributes argument on every call to save even if that will not change (I can hardly imagine an example when someone is using different consistency level for the same save method).
So should it be:
``` 
 @Insert(consistencyLevel = ANY)
    void save(Simple simple);
```
?
Also, I think that we should give our users the ability to set it on the `@Dao` level.
`@Dao(consistencyLevel = ANY)`
wdyt?

Reply:
This was considered and decided against. The ticket as the explanation as to why.

Reply:
Could you give me a link to the discussion?
In the ticket https://datastax-oss.atlassian.net/browse/JAVA-2139 I see only `Open questions:`

Reply:
[JAVA-2138](https://datastax-oss.atlassian.net/browse/JAVA-2138) will introduce method-level annotations.
I hadn't thought of overriding at the DAO level, but why not.